crates/autopilot/src/database/on_settlement_event_updater.rs

```diff
@@ -60,6 +63,35 @@ impl super::Postgres {
             )
             .await
             .context("insert_settlement_observations")?;
+
+            // update order executions for partial limit orders
+            // partial limit orders are a special kind of orders for which the surplus_fee
+            // is calculated AFTER the settlement is settled on chain.
+            for order_execution in auction_data.order_executions {
+                database::order_execution::update_surplus_fee(
+                    &mut ex,
+                    &ByteArray(order_execution.0 .0), // order uid
+                    auction_data.auction_id,
+                    Some(order_execution.1) // order fee
+                        .as_ref()
+                        .map(u256_to_big_decimal)
+                        .as_ref(),
+                )
+                .await
+                .context("insert_missing_order_executions")?;
+            }
+
+            // delete auction prices for auction_id and all auctions before it
+
+            // TODO dont delete all auctions, instead:
+            // 1. for current auction, delete all entries containing prices for tokens not
+            // included in the auction
+            // 2. for all auctions before current, if not already filtered, delete all
+            // entries (might need to add a column to auction_prices to indicate
+            // if it was filtered or not)
+            database::auction_prices::delete(&mut ex, auction_data.auction_id)
+                .await
+                .context("delete_before_auction_id")?;
```

Comment:
It feels weird that we need to delete prices here. Specifically, the driver already communicates which orders are included, so we can instead just store the prices for the orders that are included instead of all of them, then we don't need to delete auctions.

Reply:
What's the strategy to deal with solvers that settle other orders than they promised?
For example an `EIP-1271` order could become unsettlable while the solver tries to bring the solution on-chain. And possibly to cut their losses they could try to substitute the order with some other order.
Or alternatively it could make sense for a solver to continue crunching the numbers on an auction it already won hoping that it could improve the solution while it tries to bring a solution on-chain.

I think it's fine to enforce that the solution contains the orders the solver promised but AFAIR we didn't talk about that, yet. Maybe we could open an issue with all the hard requirements we need to enforce for the new API and add to it all the little things we discover while working on the new API.